meetings/2018-04-04.md

```diff
@@ -0,0 +1,143 @@
+  * several people ok for rubber stamp, Rod was pingned we’ll give him a day or 2 to 
+    review.
+  * PR author had marked as semver major
+
+* stream: 'readable' have precedence over flowing #18994
+  * Needs another TSC reviewer.  
+  * Anna will volunteer to review.
+
+#### nodejs/TSC
+* Tracking issue for updating TSC on Board Meetings #476
+  * In public session 3 topics
+    * mentorship update.
+      * node-slackers being used for meetings
+      * links to sign up have gone out.
+      * first meeting expected this week
+      * 6 month relationship, 1 hour a month
+    * leadership subcommittee
+      * update on evaluating individual membership and who is participating
+    *  JS Interactive update
+      * Nothing of interest to report on this front.
+* Strategic Initiatives - Tracking Issue #423
+  * N-API 
+    * Backports for 8.X and 6.X in progress.
+    * Plan to ramp up effort on documentation for node-addon-api.
+    * Nicola helping with a number of module backports.
+  * http2
+    * ongoing work on making sure its solid
+  * modules
+      * yesterday 8 hour online summit with presentations/discussion
+      * 15-17 members
+      *  Videos available: https://www.youtube.com/watch?v=IbPxOaGveXA&list=PLfMzBWSH11xb9LoHLO0q3Yo7HXjMOm11N
+
+  * Error messages
+  * V8 Currency
+  * Workers
+  * Core promise APIs
+  * TSC Governance
+ * New stream APIs
+   * More progress on C++ api prototype examples
+
+#### nodejs/modules
+* Upstream changes to ESM in nodejs/node #42
+  * modules team met, discussed how to handle upstream progress
+  * original request to halt, recommendation was to
+    * halt new semver minors until team has time to discuss
+    * patches/fixes can land
+  * don’t want to have people adopting pieces that won’t be supported in the end
+  * any problem with the recommendations, not formal but want TSC ok with general approach
+  * any objections?
+
+#### domenic/zones in tc39
+* discuss unblocking adding error handling #9
+  * Bradley led discussion, problem is lack of progress
+  * Oldish proposal (3 years) which are a bit like domains
+  * Domains, have some issues is problem determination, and some hooks for error handling
+  * Hooks for entering async context, subset of what domains were intended to be.
+  * There is one issue, but nothing official that Node.js is blocking zones but feeling is that
+     Node.js was against them. Bradley/Trevor were the main ones speaking up but Bradley
+     has not been able to sync with Trevor to discuss. 
+  * Key issue was that changing storage also changes error handling.
+  * Trivial to implement zones with domains, but concerns were:
+     * error propagation issues
+       * you don’t know if your unhandled exception go
+       * with promises we already have post mortem issues from domains, people
+         Just making so errors get swallowed.
+    * concerns over performance
+      * Jeremiah, different perf concerns from async_hooks.  For async_hooks only
```

Comment:
We've thought about zones for quite a bit when this was a hot topic, and I think the current proposal (although it's outdated) is fairly cheap even though it's always on. It's basically the cell to store the *current zone* plus an additional link for PromiseReactionJobs. To put this into perspective: `async_hooks` are **a lot** more expensive (on the promise side) when enabled.